boutdata/restart.py

```diff
@@ -184,19 +184,26 @@ def is_pow2(x):
                     break
 
             nx, ny, nz = data.shape
+            dx, dy, dz = old['dx'].flat[0], old['dy'].flat[0], old['dz'].flat[0]
+            # shift grid if CELL-CENTRED   
+            shift = 0.5 if old.attributes(var)['cell_location'] == 'CELL-CENTRE' else 0
```

Comment:
Need to account for possible staggering separately for each dimension.
```suggestion
            xshift = 0.0 if old.attributes(var)['cell_location'] == 'CELL_XLOW' else 0.5
            yshift = 0.0 if old.attributes(var)['cell_location'] == 'CELL_YLOW' else 0.5
            zshift = 0.0 if old.attributes(var)['cell_location'] == 'CELL_ZLOW' else 0.5
```